configtx/application.go

```diff
@@ -423,15 +431,359 @@ func (a *ApplicationOrg) setMSPConfig(updatedMSP MSP) error {
+
+	err = msp.validateCACerts()
+	if err != nil {
+		return err
+	}
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// RemoveTLSIntermediateCert removes a trusted TLS intermediate cert from the organization MSP.
+func (m *ApplicationOrgMSP) RemoveTLSIntermediateCert(cert *x509.Certificate) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	certs := msp.TLSIntermediateCerts[:]
+	for i, c := range msp.TLSIntermediateCerts {
+		if c.Equal(cert) {
+			certs = append(certs[:i], certs[i+1:]...)
+			break
+		}
+	}
+
+	msp.TLSIntermediateCerts = certs
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// SetClientOUIdentifier sets the NodeOUs client ou identifier for the organization MSP.
+func (m *ApplicationOrgMSP) SetClientOUIdentifier(clientOU membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.NodeOUs.ClientOUIdentifier = clientOU
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// SetPeerOUIdentifier sets the NodeOUs peer ou identifier for the organization MSP.
+func (m *ApplicationOrgMSP) SetPeerOUIdentifier(peerOU membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.NodeOUs.PeerOUIdentifier = peerOU
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// SetAdminOUIdentifier sets the NodeOUs admin ou identifier for the organization MSP.
+func (m *ApplicationOrgMSP) SetAdminOUIdentifier(adminOU membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.NodeOUs.AdminOUIdentifier = adminOU
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// SetOrdererOUIdentifier sets the NodeOUs orderer ou identifier for the organization MSP.
+func (m *ApplicationOrgMSP) SetOrdererOUIdentifier(ordererOU membership.OUIdentifier) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.NodeOUs.OrdererOUIdentifier = ordererOU
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// SetEnableNodeOUs sets the NodeOUs recognition, if NodeOUs recognition is enabled then an msp identity
+// that does not contain any of the specified OUs will be considered invalid.
+func (m *ApplicationOrgMSP) SetEnableNodeOUs(isEnabled bool) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.NodeOUs.Enable = isEnabled
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// AddCRL adds an CRL to the identity revocation list for the organization MSP.
+func (m *ApplicationOrgMSP) AddCRL(crl *pkix.CertificateList) error {
+	msp, err := getMSPConfig(m.configGroup)
+	if err != nil {
+		return err
+	}
+
+	msp.RevocationList = append(msp.RevocationList, crl)
+
+	return msp.setMSPConfig(m.configGroup)
+}
+
+// AddCRLFromSigningIdentity creates an crl from the provided signing identity and certs then adds the CRL to the identity revocation list for the organization MSP.
```

Comment:
`AddCRLFromSigningIdentity creates a CRL from the provided signing identity and associated certs and then adds the CRL to the identity revocation list for the organization MSP.`

Reply:
🛠️